core/sail/base/src/main/java/org/eclipse/rdf4j/sail/base/SailSourceConnection.java

```diff
@@ -135,10 +141,17 @@
 	 * @param resolver
 	 */
 	protected SailSourceConnection(AbstractSail sail, SailStore store, FederatedServiceResolver resolver) {
+		this(sail, store, new SimpleEvaluationStrategyFactory(resolver), resolver);
+	}
+
+	protected SailSourceConnection(AbstractSail sail, SailStore store,
+			EvaluationStrategyFactory evalStratFactory, FederatedServiceResolver resolver)
```

Comment:
Should it be moved up to AbstractSail/AbstractSailConnection in that case? What you are describing sounds more like TripleSource - evaluation is provided for it. In the same way that there is a SailRepository for Repository, it appears to me you are missing something that plays the role of a TripleSourceSail/EvaluationStrategySail to complete the API journey:

>
Repository - query language
   |
   |- SailRepository - Repository delegates to Sail via QueryParser
   |
Sail - query algebra
   |
   |- ??? Sail delegates to TripleSource via EvaluationStrategy
   |
TripleSource - statements

At the moment, it is on a sail impl-by-impl basis, e.g. MemoryStore and NativeStore. Maybe AbstractSail is a good candidate?